spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -256,4 +275,56 @@ public String toString() {
         runtimeFilterExpressions,
         caseSensitive());
   }
+
+  @Override
+  public CustomTaskMetric[] reportDriverMetrics() {
+    List<CustomTaskMetric> customTaskMetrics = Lists.newArrayList();
+    Optional<ScanReport> scanReportOptional = sparkReadMetricReporter.getScanReport();
```

Comment:
Iceberg historically uses null instead of `Optional` and I think we should continue to follow that. Also, Spotless makes these closures really hard to heard.

What about something like this?

```
@Override
public CustomTaskMetric[] reportDriverMetrics() {
  ScanReport scanReport = scanReportSupplier.get();

  if (scanReport == null) {
    return new CustomTaskMetric[0];
  }

  List<CustomTaskMetric> driverMetrics = Lists.newArrayList();
  driverMetrics.add(TaskTotalFileSize.from(scanReport));
  ...
  return driverMetrics.toArray(new CustomTaskMetric[0]);
}
```

Where `TaskTotalFileSize` would be defined as follows:

```
public class TaskTotalFileSize implements CustomTaskMetric {

  private final long value;

  public static TaskTotalFileSize from(ScanReport scanReport) {
    CounterResult counter = scanReport.scanMetrics().totalFileSizeInBytes();
    long value = counter != null ? counter.value() : -1;
    return new TotalFileSizeTaskMetric(value);
  }

  private TaskTotalFileSize(long value) {
    this.value = value;
  }

  @Override
  public String name() {
    return TotalFileSize.NAME;
  }

  @Override
  public long value() {
    return value;
  }
}
```